azure/hub/network_watcher.tf

```diff
@@ -3,7 +3,7 @@
 ##################
 
 resource "azurerm_resource_group" "network_watcher" {
-  count = local.enable_network_watcher ? 1 : 0
+  for_each = var.network_watcher_config.resource_group_name != null ? [0] : []
```

Comment:
Should this not be reverted? If we want a `network_watcher` resource, we have to have a `resource_group`. So therefore if `local.enable_network_watcher == true` then surely we would need a resource group. 

There is an argument to be made of whether or not we use an existing resource group but in that case, the logic on Line 18 does not account for this. Rather line 18 of this file breaks if `var.network_watcher_config.resource_group_name == null`.